server/etcdserver/api/etcdhttp/health.go

```diff
@@ -193,3 +224,135 @@ func checkAPI(lg *zap.Logger, srv ServerHealth, serializable bool) Health {
 	lg.Debug("serving /health true")
 	return h
 }
+
+type HealthCheck func(ctx context.Context) error
+
+type CheckRegistry struct {
+	path string
+
+	checks map[string]HealthCheck
+	// the list of checkNames are needed to ensure the order of checks.
```

Comment:
Why do we need to ensure order of checks?

Reply:
It was under the assumption if one of checks failed, handler return false early. This statement no longer stays so we should move it.

Reply:
As @chaochn47 said, it was added due to an old assumption. Removed.